### DISCUSSION TITLE
Jacobian in NavState::Create

### DISCUSSION THREAD

**Discussion by huanglilong** [borglab/gtsam#1080] [Q&A] [answered]
I'm very confused with jacobians (**bold text**) with translation and velocity in NavState,
where can i find docs for these. thanks a lot! 
code path: https://github.com/borglab/gtsam/blob/develop/gtsam/navigation/NavState.cpp

NavState NavState::Create(const Rot3& R, const Point3& t, const Velocity3& v,
    OptionalJacobian<9, 3> H1, OptionalJacobian<9, 3> H2,
    OptionalJacobian<9, 3> H3) {
  if (H1)
    *H1 << I_3x3, Z_3x3, Z_3x3;
  if (H2)
    *H2 << Z_3x3, **R.transpose()**, Z_3x3;
  if (H3)
    *H3 << Z_3x3, Z_3x3, **R.transpose()**;
  return NavState(R, t, v);
}

i thinks the jacobian for translation and velocity should be (I know that I was wrong)
*H2 << Z_3x3, **R.transpose()**, Z_3x3;   => *H2 << Z_3x3, I_3x3, Z_3x3
*H3 << Z_3x3, Z_3x3, **R.transpose()**   => *H2 << Z_3x3, Z_3x3, I_3x3

**Comment by dellaert** [comment, marked as answer]
math.pdf

**Reply by huanglilong** [reply]
i need more time to deep dive. Thanks a lot!

**Comment by huanglilong** [comment]
Hi @dellaert, many thanks for your suggestion last time. It's very helpful. I have gone through all the material you recommended. Would you mind help me check with the following formulas one more time? I just want to make sure what i think is correct. Thanks in advance!

Reference: https://github.com/borglab/gtsam/blob/develop/doc/ImuFactor.pdf (page 5); https://github.com/borglab/gtsam/blob/develop/doc/math.pdf

the jacobians in NavState::Create
![image](https://user-images.githubusercontent.com/7278867/152676784-d3eee4f8-f767-45c8-9870-673677b7ab25.png)

**Reply by dellaert** [reply]
No, sorry, don't have time for that. 
If you think there is an error anywhere please provide a unit test that fails.

**Reply by huanglilong** [reply]
thanks a lots!
actually, I just need help to check my derivation procession for jacobians (that same as in the code)
